sqlglot/jsonpath.py

```diff
@@ -34,6 +34,7 @@ class JSONPathTokenizer(Tokenizer):
 
     IDENTIFIER_ESCAPES = ["\\"]
     STRING_ESCAPES = ["\\"]
+    IDENTIFIERS = ["`", '"']
```

Comment:
This doesn't look right, we shouldn't have to change the JSON Path dialect, it's a separate concept.

Reply:
I agree, but how would we be able to decouple the `jsonpath.py` parsing from the Databricks specific syntax if the former can't parse the keys with special chars properly?

Reply:
It looks like `Parser._parse_colon_as_json_extract()` handles this fine, meaning that for the example:

```SQL
SELECT raw:`zip code`, raw:`fb:testid` FROM store_data
```

 it will parse the following pairs:
```
(this='raw', json_path=['`zip code`'])
(this='raw', json_path=['`fb:testid`'])
```

However, when it passes these pairs to the `exp.JsonExtract` construction the `json_path` breaks due to the `dialect.to_json_path()` call i.e because of the `jsonpath.py` tokenizer  afaict. In turn, the following incorrect query is produced:

```SQL
SELECT raw:["`zip"]["code`"], raw:["`fb"]["testid`"]
```

Reply:
Is this something that we could fix by overriding `to_json_path` for `Databricks` then?

Reply:
Probably not, `dialect.to_json_path()` is mostly a wrapper around `jsonpath.py::parse()` which in turn contains all of the JSONPath parsing under it. Sadly, the API here is not granular in that regard. 

If the `JSONPathTokenizer::IDENTIFIERS` approach is fine for Databricks but questionable for other dialects, the closest solution I can see will be to create a `databricks.py::to_json_path()` impl which will forward `IDENTIFIERS` to `jsonpath.py::parse()` or `JSONPathTokenizer`.

Reply:
I'm not sure I like either approach, need to think about this a bit. Let's sync in Slack.